### PR TITLE
OS2WEB-156: Database search

### PR DESCRIPTION
**Pull request by rimi-itk**
https://jira.itkdev.dk/browse/OS2WEB-156

### CODE REVIEW

File: modules/loop_search_db/loop_search_db.module [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+      'id' => $node->nid,
+    );
+  }
+
+  // Return as JSON.
+  drupal_json_output($data);
+}
+
+/**
+ * Implements hook_init().
```
Comment (martinyde) [inline modules/loop_search_db/loop_search_db.module]: Should this not follow the form? I.e only be loaded when the form is present? In a form alter perhaps?

File: modules/loop_search_db/loop-search-db--form.tpl.php [hunk 2]
```diff
@@ -5,5 +5,74 @@
...
+                  </div>
+                <?php endforeach ?>
+              </div>
+              <span data-ng-click="toggleFilter()" class="search-filters--action"><?php print t('Filter') ?></span>
+            </div>
+          </div>
+        <?php endif ?>
+      </div>
+
+<?php if(0===1): ?>
```
Comment (martinyde) [inline modules/loop_search_db/loop-search-db--form.tpl.php]: I think this needs a comment with an explanation

File: modules/loop_search_db/loop_search_db.module [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    return [
+      'filter' => $term->name,
+      'active' => in_array($term->name, loop_search_db_get_active_filters($facet_name), true),
+    ];
+  }, $terms);
+
+  $form['#theme'] = ['loop_search_db_form'];
+}
+
+function loop_search_db_get_active_filters($facet) {
```
Comment (martinyde) [inline modules/loop_search_db/loop_search_db.module:276]: Missing function description

**Review by martinyde — CHANGES_REQUESTED** [review]

File: modules/loop_search_db/loop_search_db.module (hunk 1 above)
Comment (rimi-itk) [reply modules/loop_search_db/loop_search_db.module]: Yes, you're absolutely right! I have to stop stealing code …

File: modules/loop_search_db/loop_search_db.module (hunk 3 above)
Comment (rimi-itk) [reply modules/loop_search_db/loop_search_db.module:276]: Will fix.

File: modules/loop_search_db/loop-search-db--form.tpl.php (hunk 2 above)
Comment (rimi-itk) [reply modules/loop_search_db/loop-search-db--form.tpl.php]: Sorting of search results were still work in progress. It has now been added.

**Review by martinyde — APPROVED** [review]

**Pull request closed without merging**